### PR TITLE
Updates for WG LC preparation:

### LINKED ISSUE #5: Clarification of layer0 label definitions
The layero-types defines groupings for augmenting generic TE label for layer0 (WSON and flexi-grid) but there is no description about the concept of layer0 label.

It may be worthwhile adding a paragraph similar to https://tools.ietf.org/html/draft-ietf-ccamp-layer1-types-06#section-4.3 for layer0 labels.

See also: https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-flexigrid-yang/pull/1/files?file-filters%5B%5D=.yang#r490981902

### LINKED ISSUE #10: Need for frequency and power range attributes also with Standard Mode
The frequency and power range attributes have not been defined for the standard mode because they are already defined in G.698.2:

https://github.com/ietf-ccamp-wg/ietf-ccamp-layer0-types-ext/pull/4#discussion_r447266914

However, after some discussion with some Q6/15 experts, it seems that G.698.2 does not mandate transceivers to be tunable nor to support tunability for the whole frequency range between the minimum and maximum frequency.

A G.698.2 application code specifies interoperability between two transceivers as long as operating on a central frequency between the minimum and maximum frequency.

A G.698.2 compliant transceiver can support, for a given application code, only one central frequency (within the range specified by G.698.2) or a limited range within the range specified by G.698.2 for that application code.

It is worth noting that operating the transceiver outside of the range specified by G.698.2 for a given application code would not be standard compliant.

Similar considerations apply for the power ranges.

Therefore it seems useful to add also to the standard mode the attributes used to report the frequency and power ranges supported by a given transceiver for a given application code.

### LINKED ISSUE #65: pyang tree length issue
Even if it is used option in the creation of tree with limit of the characters to 69, 
pyang -f tree --tree-line-length=69 ,
it seems that the result can exceed the 69 characters when it is not possible to identify where to cut the lines.
As example 
`
    |     |              +--ro max-polarization-dependent-loss-penalty* []
`

To solve this issue the attribute has been renamed with a shorter name:

`
    |     |              +--ro max-pol-dependent-loss-penalty* []
`

However, finding a more generic solution would be preferable especially when the YANG tree is quite deep

- [ ] @italobusi @sergiobelotti check with Netmod WG

### LINKED ISSUE #69: Try to shorten the names of attributes
Linked to issue #65 , the intent is to analuze the modules and try to shorten the names of attributes that appears too long, without loosing the meaning of the attribute.

### PR DESCRIPTION
**Pull request by italobusi**
- renamed decimal-16-digits and decimal-16-digits-or-null as power-spectral-density and power-spectral-density-or-null: see https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/pull/160
- added ratio-in-db and ratio-in-db-or-null: see https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/pull/160
- Shortened types and attribute names: fix #69 
  - power-spectral-density -> psd
  - power-spectral-density-or-null -> psd-or-null
  - power-in-dbm -> power-dbm
  - power-in-dbm-or-null -> power-dbm-or-null
  - decimal-2-digits -> decimal-2
  - decimal-2-digits-or-null -> decimal-2-or-null
  - decimal-5-digits -> decimal-5
  - decimal-5-digits-or-null -> decimal-5-or-null
  - ratio-in-db -> power-ratio
  - ratio-in-db-or-null -> power-ratio-or-null
  - gain-in-db -> power-gain
  - gain-in-db-or-null -> power-gain-or-null
  - loss-in-db -> power-loss
  - loss-in-db-or-null -> power-loss-or-null
- Fixed some standard references: see #6
- Added frequency and power range attributes to Standard Mode: fix #10
- Added text to clarify the WDM label description: fix #5
- Include folded tree: fix #65 

---

Co-authored-by: sergio belotti <sergio.belotti@nokia.com>

### CODE REVIEW

**Review by manzoro — APPROVED** [review]
Looks fine for me

**Comment by sergiobelotti** [comment]
> Looks fine for me

@manzoro : thanks Roberto for the review . We made some additional update related to other issues. See "additional changes" below

**Comment by sergiobelotti** [comment]
> Looks fine for me

@manzoro : thanks, remember to sign approval in the review-changes, otherwise does not result in GitHub the approval

**Review by ggalimba56 — APPROVED** [review]
I'm OK with the modifications

**Review by aguoietf — APPROVED** [review]

**Review by manzoro — APPROVED** [review]
Fine for me

File: ietf-layer0-types.yang [hunk 1]
```diff
@@ -1948,7 +2017,7 @@ module ietf-layer0-types {
         "OIF-400ZR-01.0: Implementation Agreement 400ZR";
     }
     leaf tx-polarization-power-difference {
-      type decimal-2-digits;
+      type decimal-2;
```
Comment (EstherLerouzic) [inline ietf-layer0-types.yang:2020]: use power-ratio instead ?

**Review by EstherLerouzic — COMMENTED** [review]
I have somme suggestions.

File: draft-ietf-ccamp-rfc9093-bis.md [hunk 2]
```diff
@@ -338,16 +338,76 @@ Please replace XXXX with the RFC number assigned to this document.
...
+      f = 193100.000 GHz + N x NCFG (measured in GHz)
+~~~~
+
+The WDM Label Range represents the frequency slots that are available for WDM LSPs to be set up over a given WDM Link.
+
+The WDM Label Range is defined by the label-restriction list, defined in {{!I-D.ietf-teas-rfc8776-update}}, which, for WDM, should be augmented using the l0-label-range-info grouping (for WSON only models) or the flexi-grid-label-range-info grouping (for DWDM flexible-grid only models) or the wdm-label-range-info grouping (for models that supports both WSON and DWDM flexible-grid).
+
+Each entry in the label-restriction list represents either the range of the available central wavelength values (in case of CWDM fixed-grid) or the range of the available nominal central frequencies values (in case of DWDM fixed or flexible grids): the grid-type attribute defines the type of grid for each entry of the list.
+
+In case of DWDM flexible grid, each entry in the label-restriction list represents also the range of the supported slow width values based on the following attributes, defined in {{?RFC7699}}:
```
[the remark this answers is not in the document]
Comment (sergiobelotti) [reply draft-ietf-ccamp-rfc9093-bis.md:376]: correct

File: draft-ietf-ccamp-rfc9093-bis.xml [hunk 3]
```diff
@@ -156,236 +159,650 @@ Please replace XXXX with the RFC number assigned to this document.</t>
...
+      f = 193100.000 GHz + N x NCFG (measured in GHz)
+]]></artwork></figure>
+
+<t>The WDM Label Range represents the frequency slots that are available for WDM LSPs to be set up over a given WDM Link.</t>
+
+<t>The WDM Label Range is defined by the label-restriction list, defined in <xref target="I-D.ietf-teas-rfc8776-update"/>, which, for WDM, should be augmented using the l0-label-range-info grouping (for WSON only models) or the flexi-grid-label-range-info grouping (for DWDM flexible-grid only models) or the wdm-label-range-info grouping (for models that supports both WSON and DWDM flexible-grid).</t>
+
+<t>Each entry in the label-restriction list represents either the range of the available central wavelength values (in case of CWDM fixed-grid) or the range of the available nominal central frequencies values (in case of DWDM fixed or flexible grids): the grid-type attribute defines the type of grid for each entry of the list.</t>
+
+<t>In case of DWDM flexible grid, each entry in the label-restriction list represents also the range of the supported slow width values based on the following attributes, defined in <xref target="RFC7699"/>:</t>
```
[the remark this answers is not in the document]
Comment (sergiobelotti) [reply draft-ietf-ccamp-rfc9093-bis.xml:429]: correct

File: draft-ietf-ccamp-rfc9093-bis.md [hunk 4]
```diff
@@ -338,16 +338,76 @@ Please replace XXXX with the RFC number assigned to this document.
...
+      lambda = 1471 nm + N x channel spacing (measured in nm)
+~~~~
+
+* In case of DWDM fixed-grid, defined in {{ITU-T_G.694.1}}, the frequency slot is defined by a fixed DWDM channel spacing (dwdm-ch-spc-type) and by the nominal central frequency, which is computed as described in {{?RFC6205}}:
+
+~~~~
+      f = 193100.000 GHz + N x channel spacing (measured in GHz)
+~~~~
+
+* In case of DWDM flexible-grid, defined in {{ITU-T_G.694.1}}, the frequency slot is defined by the slot width and by the nominal central frequency, which are computed, based on the slot width granularity (SWG) and of the nominal central frequency granularity (NCFG) respectively, as described in {{?RFC7699}}:
```
[the remark this answers is not in the document]
Comment (sergiobelotti) [reply draft-ietf-ccamp-rfc9093-bis.md:363]: We reference RFC7699 that cites in this way in section https://datatracker.ietf.org/doc/html/rfc7699#section-4.1:
"...Within the fixed-grid network, the C.S. value is used to represent
   the channel spacing, as the spacing between adjacent channels is
   constant.  For the flexible-grid situation, this field is used to
   represent the nominal central frequency granularity.

   This document defines a new C.S. value to supplement those in
   [[RFC6205](https://datatracker.ietf.org/doc/html/rfc6205)]:

   +------------+---------+
   | C.S. (GHz) |  Value  |
   +------------+---------+
   |     6.25   |    5    |
   +------------+---------+ ..."

File: ietf-layer0-types.yang (hunk 1 above)
Comment (sergiobelotti) [reply ietf-layer0-types.yang:2020]: it makes sense for me since power-ration is typedef of decimal-2 type

**Review by sergiobelotti — APPROVED** [review]

File: draft-ietf-ccamp-rfc9093-bis.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (italobusi) [reply draft-ietf-ccamp-rfc9093-bis.md:363]: We have modelled the NCFG and SWG as additional parameters to allow using this model also with finer vendor-specific granularity of NCFG and SWG is defined (e.g., 3.125 GHz)

File: draft-ietf-ccamp-rfc9093-bis.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (italobusi) [reply draft-ietf-ccamp-rfc9093-bis.md:376]: re-opened it not to forget to make this change before merging the PR

File: ietf-layer0-types.yang [hunk 5]
```diff
@@ -2018,54 +2087,45 @@ module ietf-layer0-types {
...
-
-/* This grouping represent the list of configured parameters */
-/* values independent of operational mode */
+  } // grouping common-all-modes
 
   grouping common-transceiver-configured-param {
-    description "Capability of an optical transceiver";
+    description
+      "The configured parameters of an optical transceiver,
+      which are independent from the configured operational mode.";
```
[the remark this answers is not in the document]
Comment (italobusi) [reply ietf-layer0-types.yang:2128]: What about "parameters which are complimentary with"?

The operational mode or application code may define the constraints of how these parameters can be set (e.g., min/max nominal central frequency or the list of allowed values for the line coding bitrate).

The constraints defined by the operational mode or application code sometimes may be strict and allow a single value to be configured (e.g., some application codes support only one line coding bitrate): in this case, the configuration of the attribute is optional and the only value allowed by the operational mode or application code can be applied as default by the system.

[the remark this answers is not in the document]
Comment (sergiobelotti) [reply ietf-layer0-types.yang:2128]: > What about "parameters which are complimentary with"?
> 
> The operational mode or application code may define the constraints of how these parameters can be set (e.g., min/max nominal central frequency or the list of allowed values for the line coding bitrate).
> 
> The constraints defined by the operational mode or application code sometimes may be strict and allow a single value to be configured (e.g., some application codes support only one line coding bitrate): in this case, the configuration of the attribute is optional and the only value allowed by the operational mode or application code can be applied as default by the system.

Gert: I would use "parameters that supplement  the configured mode ".

File: ietf-layer0-types.yang (hunk 1 above)
Comment (italobusi) [reply ietf-layer0-types.yang]: Resoved in https://github.com/ietf-ccamp-wg/ietf-ccamp-layer0-types-ext-RFC9093-bis/pull/82/commits/ad09ab3fbf5a49dec5e467d485c6099d31687db2

File: ietf-layer0-types.yang (hunk 5 above)
[the remark this answers is not in the document]
Comment (italobusi) [reply ietf-layer0-types.yang]: Resolved in https://github.com/ietf-ccamp-wg/ietf-ccamp-layer0-types-ext-RFC9093-bis/pull/82/commits/ad09ab3fbf5a49dec5e467d485c6099d31687db2

File: draft-ietf-ccamp-rfc9093-bis.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (italobusi) [reply draft-ietf-ccamp-rfc9093-bis.md]: Resolved in https://github.com/ietf-ccamp-wg/ietf-ccamp-layer0-types-ext-RFC9093-bis/pull/82/commits/ad09ab3fbf5a49dec5e467d485c6099d31687db2

**Pull request merged**